src/game/hud.cpp

```diff
@@ -1355,13 +1383,34 @@ namespace hud
                 }
                 if(crosshairhitspeed && totalmillis-game::focus->lasthit <= crosshairhitspeed)
                 {
-                    vec c2(1, 1, 1);
-                    if(hitcrosshairtone) skewcolour(c2.r, c2.g, c2.b, hitcrosshairtone);
-                    else c2 = c;
-                    drawpointertex(getpointer(POINTER_HIT, game::focus->weapselect), cx-cs/2, cy-cs/2, cs, c2.r, c2.g, c2.b, crosshairblend*hudblend);
+                    hit = true; 
+                    if (hitcrosshairtone)
+                    {
+                        skewcolour(color_hit.r, color_hit.g, color_hit.b, hitcrosshairtone);
+                    }
+                    else
+                    {
+                        if (use_custom_crosshair_colors)
+                        {
+                            color_hit = vec::hexcolor(crosshaircolorhit);
+                            // so the default crosshair will also have the hit color
+                            c = color_hit;
+                        }
+                        else
+                        {
+                            color_hit = c;
```

Comment:
better to avoid branching. `color_hit = c;` can go before the if and avoid the use of an else statement.

Reply:
true, will do